src/context/AppContext.tsx

```diff
@@ -94,23 +97,46 @@ export const AppContextProvider = ({ children }: { children: ReactNode }) => {
     }
   }, [updateUser, wallet]);
 
+  const asyncFunction = useAsyncFunction({
```

Comment:
Moved this here to allow it to be used by other components, like I have done within the `Join this Colony` button. Allowing us to prompt a user to connect, as opposed to direct them to the Colony creation page.

This is also more relevant considering the user creation happens on connection.